controlplane/kubeadm/internal/kubeadm_config_map.go

```diff
@@ -82,6 +87,33 @@ func (k *kubeadmConfig) UpdateKubernetesVersion(version string) error {
 	return nil
 }
 
+// UpdateCoreDNSImageInfo changes the dns.ImageTag and dns.ImageRepository
+// found in the kubeadm config map
+func (k *kubeadmConfig) UpdateCoreDNSImageInfo(repository, tag string) error {
```

Comment:
It feels a bit odd to bundle this into kubeadmConfig here. I'm wondering if `kubeadmConfig` should be renamed or if possibly a new type should be added and they can share a subset of implementation?

Reply:
Sounds good. We can do this as a refactor post-release. 🙂